### PR TITLE
fix: video description

### PR DESCRIPTION
**Pull request by GeronimoJohn**
# Description

### <samp>🤖 Generated by Copilot at 2010931</samp>

This pull request improves the video library components by adding a feature to show or hide the video description text. This applies to both `LocalDetails` and `YouTubeDetails` components, which are modified to use a common `TruncatedText` component. This enhances the usability and appearance of the video library.

[- Link to Basecamp Todo](https://3.basecamp.com/3105655/buckets/32318254/todos/6051672195)

# How should this PR be QA Tested?

- [ ] it should enable users to read more of the video description on More button click
- [ ] it should enable users to read less of the video description on Less button click

# Walkthrough

### <samp>🤖 Generated by Copilot at 2010931</samp>

*  Add a state variable `isExpanded` to toggle the display of video description text in `LocalDetails` and `YouTubeDetails` components ([link](https://github.com/JesusFilm/core/pull/1508/files?diff=unified&w=0#diff-21965ea058cb7a30f27547734a5418bed29f950adaa8e192c24c59dad949d84bR57), [link](https://github.com/JesusFilm/core/pull/1508/files?diff=unified&w=0#diff-179ccfbeafeafce498a27b6b14a4e5da1e0733f3e2604573f79ce0ee1bf9be51R38))
*  Use `WebkitLineClamp` CSS property to limit the number of lines shown for the video description text based on `isExpanded` state in both components ([link](https://github.com/JesusFilm/core/pull/1508/files?diff=unified&w=0#diff-21965ea058cb7a30f27547734a5418bed29f950adaa8e192c24c59dad949d84bL166-R205), [link](https://github.com/JesusFilm/core/pull/1508/files?diff=unified&w=0#diff-179ccfbeafeafce498a27b6b14a4e5da1e0733f3e2604573f79ce0ee1bf9be51L132-R172))
*  Add a `Button` component to allow the user to expand or collapse the video description text in both components, and apply some styling to create a gradient effect ([link](https://github.com/JesusFilm/core/pull/1508/files?diff=unified&w=0#diff-21965ea058cb7a30f27547734a5418bed29f950adaa8e192c24c59dad949d84bL166-R205), [link](https://github.com/JesusFilm/core/pull/1508/files?diff=unified&w=0#diff-179ccfbeafeafce498a27b6b14a4e5da1e0733f3e2604573f79ce0ee1bf9be51L132-R172))
*  Change the alignment of the buttons below the video description text in `YouTubeDetails` component to be at the end of the row ([link](https://github.com/JesusFilm/core/pull/1508/files?diff=unified&w=0#diff-179ccfbeafeafce498a27b6b14a4e5da1e0733f3e2604573f79ce0ee1bf9be51L132-R172))

### CODE REVIEW

File: apps/journeys-admin/src/components/Editor/VideoLibrary/VideoFromYouTube/YouTubeDetails/YouTubeDetails.tsx [hunk 1]
```diff
@@ -129,17 +130,46 @@ export function YouTubeDetails({
...
+            </Box>
           </Box>
         </>
       )}
-      <Stack
-        direction="row"
-        spacing={2}
-        sx={{ justifyContent: 'space-between' }}
-      >
+      <Stack direction="row" spacing={2} sx={{ justifyContent: 'end' }}>
```
Comment (tataihono) [inline apps/journeys-admin/src/components/Editor/VideoLibrary/VideoFromYouTube/YouTubeDetails/YouTubeDetails.tsx:172]: does this need to get changed?

File: apps/journeys-admin/src/components/Editor/VideoLibrary/VideoFromLocal/LocalDetails/LocalDetails.tsx [hunk 2]
```diff
@@ -163,9 +164,45 @@ export function LocalDetails({
...
-              {data?.video?.description?.find(({ primary }) => primary)?.value}
-            </Typography>
+            <Box sx={{ position: 'relative' }}>
+              <Typography
+                variant="caption"
+                sx={{
+                  display: '-webkit-box',
+                  overflow: 'hidden',
+                  WebkitLineClamp: isExpanded ? '0' : '3',
+                  WebkitBoxOrient: 'vertical',
```
Comment (tataihono) [inline apps/journeys-admin/src/components/Editor/VideoLibrary/VideoFromLocal/LocalDetails/LocalDetails.tsx:174]: what happens to this on non webkit browsers?

**Review by tataihono — DISMISSED** [review]
this solution really only works on webkit. Can we do this some other way?

File: apps/journeys-admin/src/components/Editor/VideoLibrary/VideoFromLocal/LocalDetails/LocalDetails.tsx [hunk 3]
```diff
@@ -163,9 +164,44 @@ export function LocalDetails({
...
               {data?.video?.title?.find(({ primary }) => primary)?.value}
             </Typography>
-            <Typography variant="caption">
-              {data?.video?.description?.find(({ primary }) => primary)?.value}
-            </Typography>
+            <Box sx={{ position: 'relative' }}>
+              <Typography
+                variant="caption"
+                sx={{
+                  display: '-webkit-box',
```
Comment (GeronimoJohn) [inline apps/journeys-admin/src/components/Editor/VideoLibrary/VideoFromLocal/LocalDetails/LocalDetails.tsx:171]: todo: update - don't use webkit box

**Pull request closed without merging**